src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -1436,13 +1436,25 @@ public void changed(Database.ChangeEvent event) {
 
             RevisionInternal rev = change.getAddedRevision();
 
+            // https://github.com/couchbase/couchbase-lite-android/issues/683
+            // DocumentChange of Database.ChangeEvent might not contain revision body.
+            // In case of include_docs=true, needs to load revision body
+            if (changesIncludesDocs && (rev.getBody() == null || !rev.getBody().hasContent())) {
+                EnumSet<TDContentOptions> options = getContentOptions();
+                try {
+                    rev = db.loadRevisionBody(rev, options);
+                } catch (CouchbaseLiteException e) {
+                    Log.w(Log.TAG_ROUTER, "Error loading revision body", e);
+                }
+            }
+
```

Comment:
I understand that the `rev` already contains the properties except deleted revision.

https://github.com/couchbase/couchbase-lite-ios/blob/master/Source/CBL_SQLiteStorage.m#L1646-L1650

@snej could you help confirming this?